cookbooks/composition_active/doc/composition_active.tex

```diff
@@ -0,0 +1,115 @@
+\paragraph{The active case.} The next step, of course, is to make the flow
+actually depend on the composition. After all, compositional fields are not only
+intended to indicate where material come from, but also to indicate the
+properties of this material. In general, the way to achieve this is to write
+material models where the density, viscosity, and other parameters depend on the
+composition, taking into account what the compositional fields actually denote
+(e.g., if they simply indicate the origin of material, or the concentration of
+things like olivine, perovskite, \ldots). The construction of material models is
+discussed in much greater detail in Section~\ref{sec:material-models}; we do not
+want to revisit this issue here and instead choose -- once again -- the simplest
+material model that is implemented in \aspect{}: the \texttt{simple} model.
+
+The place where we are going to hook in a compositional dependence is the
+density. In the \texttt{simple} model, the density is fundamentally described by
+a material that expands linearly with the temperature; for small density
+variations, this corresponds to a density model of the form
+$\rho(T)=\rho_0(1-\alpha(T-T_0))$. This is, by virtue of its simplicity, the
+most often considered density model. But the \texttt{simple} model also has a
+hook to make the density depend on the first compositional field $c_1(\mathbf
+x,t)$, yielding a dependence of the form
+$\rho(T)=\rho_0(1-\alpha(T-T_0))+\gamma c_1$. Here, let us choose $\rho_0=1,
+\alpha=0.01, T_0=0, \gamma=100$. The rest of our model setup will be as
+in the passive case above. Because the temperature will be between zero and one,
+the temperature induced density variations will be restricted to 0.01, whereas
+the density variation by origin of the material is 100. This should make sure
+that dense material remains at the bottom despite the fact that it is hotter
+than the surrounding material.%
+\footnote{The actual values do not matter as much here. They are chosen in such
+a way that the system -- previously driven primarily by the velocity boundary
+conditions at the top -- now also feels the impact of the density variations.
+To have an effect, the buoyancy induced by the density difference between
+materials must be strong enough to balance or at least approach the forces
+exerted by whatever is driving the velocity at the top.}
+
+
+This setup of the problem can be described using an input file that is almost
+completely unchanged from the passive case. The only difference is the use of
+the following section (the complete input file can be found in
+\url{cookbooks/composition\_active/composition\_active.prm}:
+
+\lstinputlisting[language=prmfile]{cookbooks/composition_active/doc/active.part.prm.out}
```

Comment:
The path here is modified for "composition_active" instead of "composition-active". The same goes for other \lsinputlisting .